src/System.Management.Automation/engine/PropertyAccessor.cs

```diff
@@ -56,142 +28,41 @@ internal enum PropertyScope
-        internal abstract bool GetDisablePromptToUpdateHelp();
-        internal abstract void SetDisablePromptToUpdateHelp(bool prompt);
-
-        /// <summary>
-        /// Existing Key = HKCU and HKLM\Software\Policies\Microsoft\Windows\PowerShell\UpdatableHelp
-        /// Proposed value = blank.This should be supported though
-        /// </summary>
-        /// <returns></returns>
-        internal abstract string GetDefaultSourcePath();
-        internal abstract void SetDefaultSourcePath(string defaultPath);
-
-#if UNIX
-        /// <summary>
-        /// Gets the application identity (name) to use for writing to syslog.
-        /// </summary>
-        /// <returns>
-        /// The string identity to use for writing to syslog.
-        /// <para>
-        /// The default value is 'powershell'
-        /// </para>
-        /// </returns>
-        internal abstract string GetSysLogIdentity();
-
-        /// <summary>
-        /// Gets the log level filter.
-        /// </summary>
-        /// <returns>One of the PSLevel values indicating the level to log.
-        /// <para>
-        /// The default value is PSLevel.Informational
-        /// </para>
-        /// </returns>
-        internal abstract PSLevel GetLogLevel();
-
-        /// <summary>
-        /// Gets the bitmask of the PSChannel values to log.
-        /// </summary>
-        /// <returns>
-        /// A bitmask of PSChannel.Operational and/or PSChannel.Analytic
-        /// <para>
-        /// The default value is PSChannel.Operational
-        /// </para>
-        /// </returns>
-        internal abstract PSChannel GetLogChannels();
-
-        /// <summary>
-        /// Gets the bitmask of keywords to log.
-        /// </summary>
-        /// <returns>
-        /// A bitmask of PSKeyword values.
-        /// <para>
-        /// The default value is all keywords other than UseAlwaysAnalytic
-        /// </para>
-        /// </returns>
-        internal abstract PSKeyword GetLogKeywords();
-
-#endif // UNIX
-        #endregion // Interface Methods
-    }
-
-    /// <summary>
-    /// JSON configuration file accessor
-    ///
-    /// Reads from and writes to configuration files. The values stored were
-    /// originally stored in the Windows registry.
-    /// </summary>
-    internal class JsonConfigFileAccessor : ConfigPropertyAccessor
-    {
         private string psHomeConfigDirectory;
         private string appDataConfigDirectory;
         private const string configFileName = "PowerShellProperties.json";
 
         /// <summary>
-        /// Lock used to enable multiple concurrent readers and singular write locks within a
-        /// single process.
-        /// TODO: This solution only works for IO from a single process. A more robust solution is needed to enable ReaderWriterLockSlim behavior between processes.
+        /// Lock used to enable multiple concurrent readers and singular write locks within a single process.
```

Comment:
We have `lock (s_cachedPolicies)` in Utils.cs - maybe that's enough?

Reply:
There are other settings except for the policy related ones and they may query the config file simultaneously, so this lock is needed.

Reply:
I think we could find a way to use only one global lock - if we want to have consistency, it doesn't matter how and what settings change.

Reply:
The querying for GP setting is a hot code path, especially for the script block logging, every script block execution will query for it. The lock `s_cachedPolicies` is to make sure that when multiple threads are querying for GP settings at the same time, only one thread will actually hit and read the configuration file.